libraries/AP_HAL/utility/RingBuffer.cpp

```diff
@@ -6,9 +6,14 @@
   implement a simple ringbuffer of bytes
  */
 
+static inline uint32_t align_power2(uint32_t u)
+{
+    return 1U << ((sizeof(u) * 8) - __builtin_clz(u - 1));
```

Comment:
doc says __builtin_clz has undefined behavior for '0', and bluez ring buffer impl does check for that, so it looks necessary.